training/README.md

```diff
@@ -0,0 +1,125 @@
+- `d4_train`: training mixture for T0
+- `d4_gpt_train`: training mixture for T0+
+- `d4_gpt_sglue_train`: training mixture for T0++
+
+For reproducibility, we have released an [already pre-processed version of the data](https://huggingface.co/datasets/bigscience/P3), which means you don't need to cache the data yourself. The only exception is [Story Cloze](https://cs.rochester.edu/nlp/rocstories/), which requires filling a form to get the data. Please refer to the previous SeqIO commands to cache the tasks related to Story Cloze once you have the dataset.
+
+Please also note that the module `promptsource.seqio_tasks` will be merged into the current repository. This is [WIP](https://github.com/bigscience-workshop/t-zero/issues/6).
+
+## Reproducing training in Mesh Tensorflow
+
+*Please make sure you first install the [`t5` library](https://github.com/google-research/text-to-text-transfer-transformer) and its dependencies.*
+
+Once you have pre-processed (and cached) the data, you can launch the training. We assume that you have access to TPU resources.
+
+To launch the training, a typical command is (please adjust the arguments to fit your setup and training):
+```bash
+export TPU_NAME=your_tpu_name
+export PROJECT=your_project_name
+export ZONE=your_project_zone
+export TPU_SIZE=v3-512
+
+export BUCKET=gs://your_bucket/
+export DATA_DIR="${BUCKET}/your_data_dir"
+export MODEL_DIR="${BUCKET}/your_model_dir"
+
+export MIXTURE_NAME="d4_gpt_sglue_train"
+export TRAIN_STEPS=1112200
+
+t5_mesh_transformer \
+    --tpu="${TPU_NAME}" \
+    --gcp_project="${PROJECT}" \
+    --tpu_zone="${ZONE}" \
+    --model_dir="${MODEL_DIR}"\
+    --additional_task_cache_dirs="${DATA_DIR}" \
+    --module_import="promptsource.seqio_tasks" \
+    --gin_file="dataset.gin" \
+    --gin_param="MIXTURE_NAME = '${MIXTURE_NAME}'" \
+    --gin_file="gs://t5-data/pretrained_models/t5.1.1.lm100k.xxl/operative_config.gin" \
+    --gin_param="utils.tpu_mesh_shape.tpu_topology = '${TPU_SIZE}'" \
+    --gin_param="run.train_steps = ${TRAIN_STEPS}" \
+    --gin_file="learning_rate_schedules/constant_0_001.gin" \
+    --gin_param="tokens_per_batch=1048576" \
+    --gin_param="pack_dataset.use_custom_ops = False" \
+    --gin_param="run.sequence_length = {'inputs': 1024, 'targets': 256}" \
+    --gin_param="model_info = '${MODEL_DIR}/model-info.txt'" \
+    --gin_param="mesh_train_dataset_fn.use_cached = True" \
+    --gin_param="serialize_num_microbatches.tokens_per_microbatch_per_replica = 2048" \
+    --gin_param="tpu_mesh_shape.model_parallelism = 8"
+```
+
+You can evaluate (using rank evaluation) the trained model with a similar command:
+
+```bash
+export TPU_NAME=your_tpu_name
+export PROJECT=your_project_name
+export ZONE=your_project_zone
+export TPU_SIZE=v3-128
+
+export BUCKET=gs://your_bucket/
+export DATA_DIR="${BUCKET}/your_data_dir"
+export MODEL_DIR="${BUCKET}/your_model_dir"
+
+export EVAL_MIXTURE_NAME="d4_score_eval"
```

Comment:
We should really make an eval mixture that only includes the tasks eval'd in the paper...

Reply:
FYI, we built that mixture in t5x, https://github.com/bigscience-workshop/t5x/blob/main/bigscience/gins/task.py#L19-L139 . Though maybe you were thinking of using the `csv` format in promptsource for this?

Reply:
> Though maybe you were thinking of using the `csv` format in promptsource for this?

Yes, seems to me it is the most straightforward given the code we have in promptsource, though it should be strictly equivalent.

Makes me think that it could be fun to also add commands to replicate training/evaluation in t5x @thomasw21 ? Do you want to open a pr when you get a chance?